utils/generator/generate-api.py

```diff
@@ -194,9 +208,92 @@ def func_name(self) -> str:
 
     @property
     def all_params(self):
+        return self.cached_all_params()
+
+    @lru_cache()
+    def cached_all_params(self):
+        """Builds all_params and caches it within an lru_cache()"""
         param_specs = self.spec.get("parameters", [])
+        params = [Parameter(spec) for spec in param_specs]
+
+        # If there's a request body build a list of
+        # parameters from the top-level of keys in the
+        # request body in case that body is a dictionary.
+        body_params = []
```

Comment:
This construction gets a lot more complex with request body parameters but hopefully the comments make things easier to follow.

Reply:
Maybe you can write an example case?